### ISSUE TITLE
updateWith(data) should merge with request.data

### ISSUE THREAD

**Issue by marcoscaceres** [w3c/browser-payment-api#649]
Given partial PaymentDetailsUpdate, should we object merge with whatever the sheet is showing? 

## Problem
Given: 

```JS
const details = {...}; // total, shippingOptions,  displayItems, etc. 
const request = new PaymentRequest(methods, details, options);
// later
request.addEventListener("shippingaddresschange", ev =>{
   ev.updateWith(new Promise(async resolve => {
       const total =  await calculateNewShippingCost(request.shippingAddress); 
       resolve({total}); // Partial update, just the total
   });
});
```

So, `details` initially populates the payment sheet with a `total`, `displayItems`, `shippingOptions` etc. 

Today, `ev.updateWith({total})` being called causes `displayItems`, `shippingOptions`, etc. to all get trashed - and only the total updates. Obviously, that's not great. 

## Proposal 
The proposal is that when `updateWith()` is called, we do a property spread over the original object, so old properties values are retained, and new properties are added (or nulled/emptied out).

Basically, this would be the browser code:

````JS
// `newData` what is sent in when via ev.updateWith() resolves 
sheet.addEventListener("updatepaymentsheet", ({newData}) => {
   // merge the existing data with the new data, so creates a partial update  
   const updatedData = {...sheet.getCurrentData(), ...newData};
  sheet.render(updatedData);
});
````      

cc @aestes, @rsolomakhin, @michelle

**Comment by ianbjacobs** [comment]
Can you describe a use case to help explain the issue? (I think I am missing context.)

**Comment by marcoscaceres** [comment]
@ianbjacobs, described the problem more fully above. This change is backwards compatible, because right now developers have to send in everything. 

However, we would need everyone to agree to make this change in their implementations... or it's gonna be much sadness.

**Comment by domenic** [comment]
So, I don't think this is actually a problem with the algorithm as written. For example, if you don't pass `displayItems`, what happens is that "If the displayItems member of details is present" is false in step 5 and in step 6, so _request_.[[details]].displayItems is never touched. The same goes for all the others.

Am I misreading?

**Comment by marcoscaceres** [comment]
Hmm, maybe it’s just incorrectly implemented in Chrome (and possibly Edge). Right now, Chrome definitely requires a full ‘details’ object or things vanish.

**Comment by domenic** [comment]
Sounds like a job for web platform tests ^_^

**Comment by marcoscaceres** [comment]
I’ll give the spec another read on Monday and make sure to test things more thoroughly.

**Comment by marcoscaceres** [comment]
@domenic is correct. This is invalid and was not correctly tested (by me). I'd assumed Chrome's implementation was correct. I'll [add additional tests](https://github.com/w3c/web-platform-tests/issues/9119).

**Comment by rsolomakhin** [comment]
> I'd assumed Chrome's implementation was correct.

My bad! :-\

**Comment by marcoscaceres** [comment]
Ok, created more tests https://github.com/w3c/web-platform-tests/pull/10488

@rsolomakhin, please take a looksy!

**Comment by rsolomakhin** [comment]
The tests looksy good!
